File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int counter = 0, final_counter = 0;
        for (const auto& x : s) {
            if (x == '(') counter++;
            else if (x == ')') counter--;
            final_counter = max(final_counter, counter);
        }
        return final_counter;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        s.chars().fold((0, 0), |(depth_cnt, final_cnt), c| {
            let depth_cnt = match c {
                '(' => depth_cnt + 1,
                ')' => depth_cnt - 1,
                _ => depth_cnt,
            };
            (depth_cnt, final_cnt.max(depth_cnt))
        }).1
    }
}